### PR TITLE
1096/tooltip

### LINKED ISSUE #1096: Detroit: Add Tooltips
**Acceptance Criteria**
* When I'm on the listing detail page, then I will see a tooltip next to AMI and Household Max Income
* When I click on the tooltip then I will see the note explaining AMI and Household Max Income
* When I'm on the listing detail page, then I will see a tooltip next to Confirmed by Property
* When I click on the tooltip then I will see the note explaining Confirmed by Property

**Notes**
* On hover / focus 

**Copy** 

AMI
Eligibility for regulated affordable housing is based on the area median income, or AMI. To be eligible for this housing, you often have to meet certain income requirements based on income levels relative to AMI.

Confirmed by Property
TBD

**Design**
https://www.figma.com/file/D2fNPS5Ucll8uz7hyMUQGB/bloom-public-single-listing?node-id=1838%3A19822

Hover
https://www.figma.com/file/D2fNPS5Ucll8uz7hyMUQGB/bloom-public-single-listing?node-id=1377%3A18706

Focus (style should be the same as our form elements and buttons)
https://www.figma.com/file/D2fNPS5Ucll8uz7hyMUQGB/bloom-public-single-listing?node-id=1992%3A20396

### PR DESCRIPTION
**Pull request by dominikx96**
## Issue

- Closes #1096
- Addresses #1096
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds a tooltip component that is visible in a "confirmed property" tag and info icon (units table) on a public listing page.

## How Can This Be Tested/Reviewed?

1. Open a listing with the "confirmed by property" status
2. Hover on the "confirmed by property"
3. Focus on the "confirmed by property"
4. Hover on the "info" icon next to the "AMI" header table 
5. Focus on the"info" icon next to the "AMI" header table
6. In points 2-5 you should see a tooltip

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: ui-components/src/blocks/Tooltip.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+.tooltip__element {
+  @apply w-full;
+  @apply fixed;
+  @apply bg-primary-dark;
+  @apply text-white;
+  @apply rounded;
+  @apply p-2;
+  @apply text-sm font-bold text-center;
+  @apply capitalize;
```
Comment (emilyjablonski) [inline ui-components/src/blocks/Tooltip.scss:18]: I don't think we want capitalize here based on the mock

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Lookin good! A couple things:
* I'm not able to keyboard focus on the AMI icon 
* The AMI icon is not visible on mobile view, bc of the different table views, not sure if we want it to be visible somewhere? Additionally on mobile should clicking the tooltip again dismiss it? @slowbot

File: ui-components/__tests__/blocks/Tooltip.test.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react"
```
Comment (emilyjablonski) [inline ui-components/__tests__/blocks/Tooltip.test.tsx:1]: 🥳

File: ui-components/src/blocks/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  id: string
+  text: string
+}
+
+export interface TooltipPosition {
+  top: number
+  left: number
+}
+
+const Tooltip: React.FC<TooltipProps> = ({ className, id, text, children }) => {
```
Comment (emilyjablonski) [inline ui-components/src/blocks/Tooltip.tsx:16]: We don't use the `React.FC` notation, so let's use the generic functional component notation

File: ui-components/src/tables/GroupedTable.tsx [hunk 4]
```diff
@@ -17,7 +18,14 @@ export const GroupedTable = (props: GroupedTableProps) => {
 
   const headerLabels = Object.values(headers).map((col, index) => {
     const uniqKey = process.env.NODE_ENV === "test" ? `header-${index}` : nanoid()
-    return <th key={uniqKey}>{col}</th>
+    return (
+      <>
```
Comment (emilyjablonski) [inline ui-components/src/tables/GroupedTable.tsx:22]: We can remove this fragment

**Comment by dominikx96** [comment]
@emilyjablonski thanks for the review! 

```
The AMI icon is not visible on mobile view, bc of the different table views, not sure if we want it to be visible somewhere? Additionally on mobile should clicking the tooltip again dismiss it? @slowbot
```

It needs a table component refactor so before I start doing it, please confirm that we need it on mobile too :)

**Comment by emilyjablonski** [comment]
Waiting on feedback from @slowbot before making any changes 👍

**Review by emilyjablonski — APPROVED** [review]
https://user-images.githubusercontent.com/65367387/165565998-58c9866b-9d53-4e3a-8582-24abcbd452e0.mov

This looks great! The scroll bug above is my only remaining question, but I think it might be fine to merge over and handle in a followup, I'll check in and let you know

**Review by YazeedLoonat — COMMENTED** [review]
Not sure if this should block merging this in, right now on mobile the `Coming Soon` Tooltip pops up when you click on "coming soon" which flashes the tooltip before being redirected away 
I think we do want to remove the tooltip for that on mobile

**Comment by emilyjablonski** [comment]
@YazeedLoonat That seems to only happen on the directory page, I'm personally good to handle it in a followup but happy either way

**Pull request merged**